tmol/pose/pose_stack_builder.py

```diff
@@ -0,0 +1,1964 @@
+            block_coord_offset64=i64(block_coord_offset),
+            inter_residue_connections=inter_residue_connections,
+            inter_residue_connections64=i64(inter_residue_connections),
+            inter_block_bondsep=inter_block_bondsep,
+            inter_block_bondsep64=i64(inter_block_bondsep),
+            block_type_ind=block_type_ind,
+            block_type_ind64=i64(block_type_ind),
+            device=device,
+        )
+
+    @classmethod
+    @validate_args
+    def from_poses(
+        cls, pose_stacks: List[PoseStack], device: torch.device
+    ) -> PoseStack:
+        pbt0 = pose_stacks[0].packed_block_types
+        reuse_pbt = all(
+            pose_stack.packed_block_types is pbt0 for pose_stack in pose_stacks
+        )
+        if reuse_pbt:
+            packed_block_types = pbt0
+        else:
+            all_bt = [
+                bt
+                for pose_stack in pose_stacks
+                for bt in pose_stack.packed_block_types.active_block_types
+            ]
+            bt_set = {}
+            for bt in all_bt:
+                if bt.name not in bt_set:
+                    bt_set[bt.name] = bt
+            uniq_bt = [v for _, v in bt_set.items()]
+            packed_block_types = PackedBlockTypes.from_restype_list(uniq_bt, device)
+
+        max_n_blocks = max(pose_stack.max_n_blocks for pose_stack in pose_stacks)
+        coords, block_coord_offset = cls._pack_pose_stack_coords(
+            packed_block_types, pose_stacks, max_n_blocks, device
+        )
+
+        ps_offset = exclusive_cumsum1d(
+            torch.tensor([len(ps) for ps in pose_stacks], dtype=torch.int64)
+        )
+        # block_coord_offset_cpu = block_coord_offset.cpu()
+
+        inter_residue_connections = cls._inter_residue_connections_from_pose_stacks(
+            packed_block_types, pose_stacks, ps_offset, max_n_blocks, device
+        )
+        inter_block_bondsep = cls._interblock_bondsep_from_pose_stacks(
+            packed_block_types, pose_stacks, ps_offset, max_n_blocks, device
+        )
+        block_type_ind = cls._resolve_block_type_ind(
+            packed_block_types, pose_stacks, ps_offset, max_n_blocks, device
+        )
+
+        def i64(t):
+            return t.to(torch.int64)
+
+        return PoseStack(
+            packed_block_types=packed_block_types,
+            coords=coords,
+            block_coord_offset=block_coord_offset,
+            block_coord_offset64=i64(block_coord_offset),
+            inter_residue_connections=inter_residue_connections,
+            inter_residue_connections64=i64(inter_residue_connections),
+            inter_block_bondsep=inter_block_bondsep,
+            inter_block_bondsep64=i64(inter_block_bondsep),
+            block_type_ind=block_type_ind,
+            block_type_ind64=i64(block_type_ind),
+            device=device,
+        )
+
+    @classmethod
+    @validate_args
+    def pose_stack_from_monomer_polymer_sequences(
```

Comment:
Can the three functions here be combined into a single function instead?  e.g. pose_stack_from_sequences?  And defaulted arguments can decide if it has breaks, non-polymeric connections, etc?